forc-plugins/forc-client/tests/deploy.rs

```diff
@@ -405,3 +410,157 @@ fn test_deploy_interactive_wrong_password() -> Result<(), rexpect::error::Error>
     node.kill().unwrap();
     Ok(())
 }
+
+#[tokio::test]
+async fn chunked_deploy() {
+    let (mut node, port) = run_node();
+    let tmp_dir = tempdir().unwrap();
+    let project_dir = test_data_path().join("big_contract");
+    copy_dir(&project_dir, tmp_dir.path()).unwrap();
+    patch_manifest_file_with_path_std(tmp_dir.path()).unwrap();
+
+    let pkg = Pkg {
+        path: Some(tmp_dir.path().display().to_string()),
+        ..Default::default()
+    };
+
+    let node_url = format!("http://127.0.0.1:{}/v1/graphql", port);
+    let target = NodeTarget {
+        node_url: Some(node_url),
+        target: None,
+        testnet: false,
+    };
+    let cmd = cmd::Deploy {
+        pkg,
+        salt: Some(vec![format!("{}", Salt::default())]),
+        node: target,
+        default_signer: true,
+        ..Default::default()
+    };
+    let deployed_contract = deploy(cmd).await.unwrap().remove(0);
+    node.kill().unwrap();
+
+    let num_chunks = deployed_contract.chunks.len();
+    assert_eq!(num_chunks, 4);
+}
+
+#[tokio::test]
+async fn chunked_deploy_re_routes_call() {
+    let (mut node, port) = run_node();
+    let tmp_dir = tempdir().unwrap();
+    let project_dir = test_data_path().join("big_contract");
+    copy_dir(&project_dir, tmp_dir.path()).unwrap();
+    patch_manifest_file_with_path_std(tmp_dir.path()).unwrap();
+
+    let pkg = Pkg {
+        path: Some(tmp_dir.path().display().to_string()),
+        ..Default::default()
+    };
+
+    let node_url = format!("http://127.0.0.1:{}/v1/graphql", port);
+    let target = NodeTarget {
+        node_url: Some(node_url.clone()),
+        target: None,
+        testnet: false,
+    };
+    let cmd = cmd::Deploy {
+        pkg,
+        salt: Some(vec![format!("{}", Salt::default())]),
+        node: target,
+        default_signer: true,
+        ..Default::default()
+    };
+    let deployed_contract = deploy(cmd).await.unwrap().remove(0);
+
+    let provider = Provider::connect(&node_url).await.unwrap();
+    let secret_key = SecretKey::from_str(forc_client::constants::DEFAULT_PRIVATE_KEY).unwrap();
+    let wallet_unlocked = WalletUnlocked::new_from_private_key(secret_key, Some(provider));
+
+    abigen!(Contract(
+        name = "BigContract",
+        abi = "forc-plugins/forc-client/test/data/big_contract/big_contract-abi.json"
+    ));
+
+    let instance = BigContract::new(deployed_contract.id, wallet_unlocked);
+    let chunks: Vec<Bech32ContractId> = deployed_contract
+        .chunks
+        .iter()
+        .cloned()
+        .map(|chunk_id| chunk_id.into())
+        .collect();
+
+    // result should be true.
+    let result = instance
+        .methods()
+        .test_function()
+        .with_contract_ids(&chunks)
```

Comment:
We should be using the new [blob mode](https://github.com/FuelLabs/fuel-vm/pull/780/files) for `ldc` so that the contract IDs don't have to be provided as inputs to every transaction.